app/models/train.rb

```diff
@@ -7,16 +7,16 @@ class Train < ApplicationRecord
 
   has_many :carriages, dependent: :nullify
 
-  def count_carraiage_types
-    carriages.each_with_object(Hash.new(0)) do |carriage, hsh|
-      hsh[carriage.carriage_type.name] += 1
-    end
+  def count_carraiage_places
+    query = 'type, SUM(bottom_places), SUM(upper_places), SUM(side_upper_places), SUM(side_bottom_places), SUM(seats)'
+    carriages.group(:type).pluck(Arel.sql(query))
   end
 
-  def count_carraiage_places
-    CarriageType.joins(:carriages)
-                .where(carriages: { train_id: carriages })
-                .group(:name)
-                .pluck(Arel.sql('carriage_types.name, SUM(bottom_places), SUM(upper_places)'))
+  def carriages_head
+    carriages.order(:number)
```

Comment:
It looks like hardcore simplification or magic please explain.

Reply:
I try do this in scope, but can not call on instance. I define method, that can call on instance.

Reply:
Oh, sorry it is my mistake, I mean where you move request?

CarriageType.joins(:carriages)
                .where(carriages: { train_id: carriages })
                .group(:name)
                .pluck(Arel.sql('carriage_types.name, SUM(bottom_places), SUM(upper_places)'))